packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -124,6 +129,13 @@ export interface SecretAttributes {
    * The ARN of the secret in SecretsManager.
    */
   readonly secretArn: string;
+
+  /**
+   * The name of the secret in SecretsManager.
+   *
+   * @default - the name is derived from the secretArn.
+   */
+  readonly secretName?: string;
```

Comment:
Why is this useful if `secretArn` is required?

Reply:
That's a fair comment; I think it's primarily a hold-over from when `fromSecretName` used `fromSecretAttributes`, which it no longer does. I kept it for completeness, but it could just as easily be removed.

Reply:
https://github.com/aws/aws-cdk/pull/10410